### PR TITLE
Add testing reference section + re-write guide

### PR DESCRIPTION
**Pull request by clrcrl**
## To-do before merge:
- [x] Finish off reference sections
- [x] Add FAQs

## Not part of this PR:
- Re-writing the documentation section

### CODE REVIEW

**Comment by drewbanin** [comment]
i think they look alright tucked inside of the list of general properties!

**Comment by clrcrl** [comment]
So I rebased this branch off master before pushing it, but something is up with the way it's rendering:
- [deploy preview](https://deploy-preview-173--docs-getdbt-com.netlify.app/docs/building-a-dbt-project/using-sources)
- [current docs](https://docs.getdbt.com/docs/building-a-dbt-project/using-sources)

No idea what this might be!

**Comment by drewbanin** [comment]
@clrcrl you might need to clear the cache and rebuild in netlify. I think this is related to the docusaurus upgrade we merged the other day

File: website/docs/docs/running-a-dbt-project/command-line-interface/model-selection-syntax.md [hunk 1]
```diff
@@ -106,3 +106,153 @@ dbt provides an `--exclude` flag with the same semantics as `--models`. Models s
...
+        tests:
+        tags: [my_column_tag]
+          - unique
+
+```
+
+</File>
+
+```shell
+$ dbt test --tags tag:my_column_tag
```
Comment (drewbanin) [inline website/docs/docs/running-a-dbt-project/command-line-interface/model-selection-syntax.md:227]:
```suggestion
$ dbt test --models tag:my_column_tag
```

File: website/docs/reference/resource-properties/tags.md [hunk 2]
```diff
@@ -29,6 +25,7 @@ version: 2
 
 models:
   - name: model_name
+    # tags: [] # check out [these docs](resource-configs/tags) for more info
```
Comment (drewbanin) [inline website/docs/reference/resource-properties/tags.md:28]: FYI the markdown link rendered doesn't like this:
https://deploy-preview-173--docs-getdbt-com.netlify.app/reference/resource-properties/tags

File: website/docs/docs/building-a-dbt-project/documentation.md [hunk 3]
```diff
@@ -1,8 +1,91 @@
...
+To declare a docs block, use the jinja `docs` tag. Docs blocks must be uniquely named, and can contain arbitrary markdown. In practice, a docs block might look like this:
+
+<File name='events.md'>
+
+```markdown
+{% docs table_events %}
+
+This table contains clickstream events from the marketing website.
+
+The events in this table are recorded by [Snowplow](http://github.com/snowplow/snowplow) and piped into the warehouse on an hourly basis. The following pages of the marketing site are tracked:
```
[the remark this answers is not in the document]
Comment (clrcrl) [reply website/docs/docs/building-a-dbt-project/documentation.md:48]: Oh thanks, I haven't touched much of this section yet, need to re-write!

[the remark this answers is not in the document]
Comment (clrcrl) [reply website/docs/docs/building-a-dbt-project/documentation.md]: Oh thanks, I haven't touched much of this section yet, need to re-write!

**Pull request merged**